enigma/src/main/java/org/quiltmc/enigma/impl/translation/mapping/MappingsChecker.java

```diff
@@ -84,26 +87,42 @@ public Dropped dropEmptyMappings(ProgressListener progress) {
 	}
 
 	private void tryDropEmptyEntry(Dropped dropped, Entry<?> entry) {
-		if (this.shouldDropEmptyMapping(entry)) {
+		if (this.shouldDropEmptyMapping(dropped, entry)) {
 			EntryMapping mapping = this.mappings.get(entry);
 			if (mapping != null) {
 				dropped.drop(entry, mapping);
 			}
 		}
 	}
 
-	private boolean shouldDropEmptyMapping(Entry<?> entry) {
+	private boolean shouldDropEmptyMapping(Dropped dropped, Entry<?> entry) {
 		EntryMapping mapping = this.mappings.get(entry);
 		if (mapping != null) {
-			boolean isEmpty = mapping.targetName() == null && mapping.javadoc() == null;
+			boolean isEmpty = (mapping.targetName() == null && mapping.javadoc() == null) || !this.project.isRenamable(entry);
+
 			if (isEmpty) {
-				return this.mappings.getChildren(entry).isEmpty();
+				return this.hasChildren(entry, dropped);
 			}
 		}
 
 		return false;
 	}
 
+	private boolean hasChildren(Entry<?> entry, Dropped dropped) {
+		var children = this.mappings.getChildren(entry);
+
+		// account for child mappings that have been dropped already
+		if (!children.isEmpty()) {
+			for (Entry<?> child : children) {
+				if (!dropped.getDroppedMappings().containsKey(child) && !this.hasChildren(child, dropped)) {
+					return true;
+				}
+			}
+		}
+
+		return children.isEmpty();
```

Comment:
I feel like the name of this method is the opposite of what this [line] does?

Reply:
oh you're so right lol